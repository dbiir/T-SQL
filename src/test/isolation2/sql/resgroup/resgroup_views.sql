select *
  from gp_toolkit.gp_resgroup_config
 where groupname='default_group';

select rsgname
     , groupid
     , num_running
     , num_queueing
     , num_queued
     , num_executed
     , cpu_usage->'-1' as qd_cpu_usage
     , memory_usage->'-1'->'used' as qd_memory_used
     , memory_usage->'-1'->'shared_used' as qd_memory_shared_used
  from gp_toolkit.gp_resgroup_status
 where rsgname='default_group';

select rsgname
     , groupid
     , cpu
     , memory_used
     , memory_shared_used
  from gp_toolkit.gp_resgroup_status_per_host s
  join gp_segment_configuration c
    on s.hostname=c.hostname and c.content=-1 and role='p'
 where rsgname='default_group';

select rsgname
     , groupid
     , segment_id
     , cpu
     , memory_used
     , memory_shared_used
  from gp_toolkit.gp_resgroup_status_per_segment
 where rsgname='default_group'
   and segment_id=-1;

-- also log the raw output of the views, if any of above tests failed it is
-- easier to find out the causes with these logs.

-- start_ignore
select * from gp_toolkit.gp_resgroup_config;
select * from gp_toolkit.gp_resgroup_status;
select * from gp_toolkit.gp_resgroup_status_per_host;
select * from gp_toolkit.gp_resgroup_status_per_segment;
-- end_ignore
