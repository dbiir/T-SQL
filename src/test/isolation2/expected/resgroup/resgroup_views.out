select * from gp_toolkit.gp_resgroup_config where groupname='default_group';
 groupid | groupname     | concurrency | cpu_rate_limit | memory_limit | memory_shared_quota | memory_spill_ratio | memory_auditor | cpuset 
---------+---------------+-------------+----------------+--------------+---------------------+--------------------+----------------+--------
 6437    | default_group | 20          | 30             | 30           | 80                  | 10                 | vmtracker      | -1     
(1 row)

select rsgname , groupid , num_running , num_queueing , num_queued , num_executed , cpu_usage->'-1' as qd_cpu_usage , memory_usage->'-1'->'used' as qd_memory_used , memory_usage->'-1'->'shared_used' as qd_memory_shared_used from gp_toolkit.gp_resgroup_status where rsgname='default_group';
 rsgname       | groupid | num_running | num_queueing | num_queued | num_executed | qd_cpu_usage | qd_memory_used | qd_memory_shared_used 
---------------+---------+-------------+--------------+------------+--------------+--------------+----------------+-----------------------
 default_group | 6437    | 0           | 0            | 0          | 0            | 0.00         | 0              | 0                     
(1 row)

select rsgname , groupid , cpu , memory_used , memory_shared_used from gp_toolkit.gp_resgroup_status_per_host s join gp_segment_configuration c on s.hostname=c.hostname and c.content=-1 and role='p' where rsgname='default_group';
 rsgname       | groupid | cpu  | memory_used | memory_shared_used 
---------------+---------+------+-------------+--------------------
 default_group | 6437    | 0.00 | 0           | 0                  
(1 row)

select rsgname , groupid , segment_id , cpu , memory_used , memory_shared_used from gp_toolkit.gp_resgroup_status_per_segment where rsgname='default_group' and segment_id=-1;
 rsgname       | groupid | segment_id | cpu  | memory_used | memory_shared_used 
---------------+---------+------------+------+-------------+--------------------
 default_group | 6437    | -1         | 0.00 | 0           | 0                  
(1 row)

-- also log the raw output of the views, if any of above tests failed it is
-- easier to find out the causes with these logs.

-- start_ignore
select * from gp_toolkit.gp_resgroup_config;
 groupid | groupname     | concurrency | cpu_rate_limit | memory_limit | memory_shared_quota | memory_spill_ratio | memory_auditor | cpuset 
---------+---------------+-------------+----------------+--------------+---------------------+--------------------+----------------+--------
 6437    | default_group | 20          | 30             | 30           | 80                  | 10                 | vmtracker      | -1     
 6438    | admin_group   | 2           | 10             | 10           | 80                  | 10                 | vmtracker      | -1     
(2 rows)
select * from gp_toolkit.gp_resgroup_status;
 rsgname       | groupid | num_running | num_queueing | num_queued | num_executed | total_queue_duration | cpu_usage                                 | memory_usage                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
---------------+---------+-------------+--------------+------------+--------------+----------------------+-------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 default_group | 6437    | 0           | 0            | 0          | 0            | @ 0                  | {"-1":0.00, "0":0.00, "1":0.00, "2":0.00} | {"-1":{"used":0, "available":204, "quota_used":0, "quota_available":40, "quota_granted":40, "quota_proposed":40, "shared_used":0, "shared_available":164, "shared_granted":164, "shared_proposed":164}, "0":{"used":0, "available":204, "quota_used":0, "quota_available":40, "quota_granted":40, "quota_proposed":40, "shared_used":0, "shared_available":164, "shared_granted":164, "shared_proposed":164}, "1":{"used":0, "available":204, "quota_used":0, "quota_available":40, "quota_granted":40, "quota_proposed":40, "shared_used":0, "shared_available":164, "shared_granted":164, "shared_proposed":164}, "2":{"used":0, "available":204, "quota_used":0, "quota_available":40, "quota_granted":40, "quota_proposed":40, "shared_used":0, "shared_available":164, "shared_granted":164, "shared_proposed":164}} 
 admin_group   | 6438    | 1           | 0            | 0          | 19           | @ 0                  | {"-1":0.41, "0":0.18, "1":0.14, "2":0.10} | {"-1":{"used":0, "available":68, "quota_used":6, "quota_available":6, "quota_granted":12, "quota_proposed":12, "shared_used":0, "shared_available":56, "shared_granted":56, "shared_proposed":56}, "0":{"used":0, "available":68, "quota_used":6, "quota_available":6, "quota_granted":12, "quota_proposed":12, "shared_used":0, "shared_available":56, "shared_granted":56, "shared_proposed":56}, "1":{"used":0, "available":68, "quota_used":6, "quota_available":6, "quota_granted":12, "quota_proposed":12, "shared_used":0, "shared_available":56, "shared_granted":56, "shared_proposed":56}, "2":{"used":0, "available":68, "quota_used":6, "quota_available":6, "quota_granted":12, "quota_proposed":12, "shared_used":0, "shared_available":56, "shared_granted":56, "shared_proposed":56}}                     
(2 rows)
select * from gp_toolkit.gp_resgroup_status_per_host;
 rsgname       | groupid | hostname      | cpu  | memory_used | memory_available | memory_quota_used | memory_quota_available | memory_shared_used | memory_shared_available 
---------------+---------+---------------+------+-------------+------------------+-------------------+------------------------+--------------------+-------------------------
 admin_group   | 6438    | nyu-vm-ubuntu | 0.83 | 1           | 271              | 24                | 24                     | 0                  | 224                     
 default_group | 6437    | nyu-vm-ubuntu | 0.00 | 0           | 816              | 0                 | 160                    | 0                  | 656                     
(2 rows)
select * from gp_toolkit.gp_resgroup_status_per_segment;
 rsgname       | groupid | hostname      | segment_id | cpu  | memory_used | memory_available | memory_quota_used | memory_quota_available | memory_shared_used | memory_shared_available 
---------------+---------+---------------+------------+------+-------------+------------------+-------------------+------------------------+--------------------+-------------------------
 admin_group   | 6438    | nyu-vm-ubuntu | 0          | 0.13 | 0           | 68               | 6                 | 6                      | 0                  | 56                      
 default_group | 6437    | nyu-vm-ubuntu | 1          | 0.00 | 0           | 204              | 0                 | 40                     | 0                  | 164                     
 default_group | 6437    | nyu-vm-ubuntu | 0          | 0.00 | 0           | 204              | 0                 | 40                     | 0                  | 164                     
 default_group | 6437    | nyu-vm-ubuntu | -1         | 0.00 | 0           | 204              | 0                 | 40                     | 0                  | 164                     
 default_group | 6437    | nyu-vm-ubuntu | 2          | 0.00 | 0           | 204              | 0                 | 40                     | 0                  | 164                     
 admin_group   | 6438    | nyu-vm-ubuntu | 1          | 0.10 | 0           | 68               | 6                 | 6                      | 0                  | 56                      
 admin_group   | 6438    | nyu-vm-ubuntu | -1         | 0.26 | 1           | 67               | 6                 | 6                      | 0                  | 56                      
 admin_group   | 6438    | nyu-vm-ubuntu | 2          | 0.10 | 0           | 68               | 6                 | 6                      | 0                  | 56                      
(8 rows)
-- end_ignore
